python_modules/dagster/dagster/core/storage/runs/base.py

```diff
@@ -43,7 +49,11 @@ def handle_run_event(self, run_id: str, event: DagsterEvent):
 
     @abstractmethod
     def get_runs(
-        self, filters: PipelineRunsFilter = None, cursor: str = None, limit: int = None
+        self,
+        filters: PipelineRunsFilter = None,
+        cursor: str = None,
+        limit: int = None,
+        bucket: Optional[Union[JobBucket, TagBucket]] = None,
```

Comment:
does it make sense to pass `limit` and `bucket` ?

if not thoughts on just having one arg that takes `Union[int, JobBucket, TagBucket]` or error when both passed?

Reply:
Yeah, I initially had things under a single arg, but I think it did get a little confusing.  I'm keeping them separate but then raising an error if both are provided.

Reply:
`limit_by` `bucket_by` `bucket_with` `limit_bucket` 

not sure i like these better just throwing stuff out there since I dont love `bucket`